runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -184,31 +185,34 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
             }
         }
 
-        // Perform underlying access
-        for (int x = 0; x < rt.getParameters().getTrimRetry(); x++) {
-            // Linearize this read against a timestamp
-            final long timestamp = rt.getSequencerView()
-                            .query(getStreamID());
-            log.debug("Access[{}] conflictObj={} version={}", this, conflictObject, timestamp);
+        // Linearize this read against a timestamp
+        AtomicLong timestamp = new AtomicLong(rt.getSequencerView().query(getStreamID()));
 
-            try {
-                return underlyingObject.access(o -> o.getVersionUnsafe() >= timestamp
-                                && !o.isOptimisticallyModifiedUnsafe(),
-                        o -> o.syncObjectUnsafe(timestamp),
-                        o -> accessMethod.access(o));
-            } catch (TrimmedException te) {
-                log.info("accessInner: Encountered trimmed address space " +
-                    "while accessing version {} of stream {} on attempt {}",
-                    timestamp, getStreamID(), x);
-                // We encountered a TRIM during sync, reset the object
-                underlyingObject.update(o -> {
-                    o.resetUnsafe();
-                    return null;
-                });
-            }
-        }
+        log.debug("Access[{}] conflictObj={} version={}", this, conflictObject, timestamp);
 
-        throw new TrimmedException();
+        // Perform underlying access
```

Comment:
Btw, `getUpcallResultInner` has the same problem. The reset in the catch block should be part of the update lambda to be executed under the same lock.

Reply:
Good catch, changed it, please take a look.